packages/common-all/src/store/FuseMetadataStore.ts

```diff
@@ -0,0 +1,232 @@
+import Fuse from "fuse.js";
+import _ from "lodash";
+import { ERROR_SEVERITY, ERROR_STATUS } from "../constants";
+import { DendronError } from "../error";
+import { FuseEngine } from "../FuseEngine";
+import { NotePropsByFnameDict, NotePropsMeta, RespV3 } from "../types";
+import { FindNoteOpts } from "../types/FindNoteOpts";
+import { IDataQueryable } from "./IDataQueryable";
+import { IDataStore } from "./IDataStore";
+import {
+  ok,
+  Ok,
+  err,
+  Err,
+  Result,
+  okAsync,
+  errAsync,
+  ResultAsync,
+  fromThrowable,
+  fromPromise,
+  fromSafePromise,
+} from "neverthrow";
+
+export class FuseMetadataStore
+  implements
+    IDataStore<string, NotePropsMeta>, // TODO: Instead of NotePropsMeta, use a generic so this also works with Schemas
+    IDataQueryable<string, NotePropsMeta[]>
+{
+  private notesIndex: Fuse<NotePropsMeta>;
+
+  private createFuse<T>(
+    initList: T[],
+    opts: Fuse.IFuseOptions<T> & {
+      preset: "schema" | "note";
+    },
+    index?: Fuse.FuseIndex<T>
+  ) {
+    const options: Fuse.IFuseOptions<T> = {
+      shouldSort: true,
+      threshold: opts.threshold,
+      distance: 15,
+      minMatchCharLength: 1,
+      keys: ["fname", "id", "vault"],
+      useExtendedSearch: true,
+      includeScore: true,
+      // As long as we have ignoreLocation set to true location the location
+      // value should be ignored.
+      location: 0,
+      ignoreLocation: true,
+      ignoreFieldNorm: true,
+      ...opts,
+    };
+    if (opts.preset === "schema") {
+      options.keys = ["fname", "id"];
+    }
+    const fuse = new Fuse(initList, options, index);
+    return fuse;
+  }
+
+  private readonly threshold: number;
+
+  constructor() {
+    // this.threshold =
+    // opts.mode === "exact" ? 0.0 : getCleanThresholdValue(opts.fuzzThreshold);
+
+    this.threshold = 0.2; // JYTODO
+
+    this.notesIndex = this.createFuse<NotePropsMeta>([], {
+      preset: "note",
+      threshold: this.threshold,
+    });
+  }
+
+  /**
+   * See {@link IDataStore.get}
+   */
+  async get(key: string): Promise<RespV3<NotePropsMeta>> {
+    const result = this.notesIndex.search({ id: `=${key}` });
+    // const maybeNote = this._noteMetadataById[key];
+
+    if (result && result.length === 1) {
+      return { data: _.cloneDeep(result[0].item) };
+    } else {
+      return {
+        error: DendronError.createFromStatus({
+          status: ERROR_STATUS.CONTENT_NOT_FOUND,
+          message: `NoteProps metadata not found for key ${key}.`,
+          severity: ERROR_SEVERITY.MINOR,
+        }),
+      };
+    }
+  }
+
+  /**
+   * See {@link IDataStore.find}
+   */
+  async find(opts: FindNoteOpts): Promise<RespV3<NotePropsMeta[]>> {
+    if (!opts.fname) {
+      return { data: [] }; // JYTODO
+    }
+
+    const searchResults = this.notesIndex.search({
+      fname: `=${opts.fname}`,
```

Comment:
@kevinslin as a heads up - I've done zero testing on these search strings; just put something in here after reading fuse documentation